File: Ch1-Fundamentals/0019-Values-Greater-than-Second/main.py
def get_values_greater_than_2nd():
    arr = [1, 3, 5, 7, 9, 13]
    
    if len(arr) == 0:
        return "\n!!! ERROR: There are no values in this array."
    
    if not all(isinstance(x, (int, float)) for x in arr):
        return "\n!!! ERROR: The array must consist of numbers."
    
    second_val = arr[1]
    count = 0
    for num in arr:
        if num > second_val:
            print(f"Value > second_val: {num}")
            count += 1
    
    return f"--------------\nValues greater than second value: {count}\n"

print(get_values_greater_than_2nd())

File: Ch1-Fundamentals/0019-Values-Greater-than-Second/main.js
function isNumericArray(arr) {
    for (let i = 0; i < arr.length; i++) {
        if (typeof arr[i] !== 'number') {
            return false;
        }
    }
    return true;
}

function valuesGreaterThanSecond() {
    const arr = [1, 3, 5, 7, 9, 13];
    
    if (arr.length === 0) {
        return "\n!!! ERROR: There are no values in this array.";
    }
    
    if (isNumericArray(arr) === true) {
        let secondVal = arr[1];
        let count = 0;
        
        for (let i = 0; i < arr.length; i++) {
            if (arr[i] > secondVal) {
                console.log("Value > secondVal: " + arr[i]);
                count++;
            }
            // if not > 2nd val, move on to the next
        }
        
        return `--------------\nValues greater than second value: ${count}\n`;
    } else {
        return "\n!!! ERROR: The array must consist of numbers.";
    }
}

console.log(valuesGreaterThanSecond());
